src/disco/FixedWidthField/Integer/src/parseInteger.hpp

```diff
@@ -32,7 +32,7 @@ parseInteger( Iterator& it, uint16_t& position, Signed ){
 
   const auto magnitude = parseInteger( it, position, Unsigned() );
 
-  if ( unlikely( magnitude > std::numeric_limits< int64_t >::max() ) ){
+  if ( unlikely( magnitude > (uint64_t) std::numeric_limits< int64_t >::max() ) ){
```

Comment:
Why can't we just do:
```suggestion
  if ( unlikely( magnitude > std::numeric_limits< uint64_t >::max() ) ){
```

Reply:
That changes the meaning of the comparison.  `std::numeric_limits< uint64_t >::max()` is twice as large as the original.  I think the appropriate comparison is to the signed integer maximum, as the actual value is signed, this just stripped away the sign.

Reply:
This is for parsing a SIGNED integer. The sign is read from the string and then an UNSIGNED int is parsed using the parseInteger function for that integer type. As such, magnitude is an unsigned int. However, to be sure we can store the value in a SIGNED integer, we need to compare to the maximum value of a SIGNED value.

Using a static_cast of the signed max value to an unsigned value is therefore the correct approach, otherwise you can get into potential overflow issues.